src/item_pocket.cpp

```diff
@@ -1914,10 +1914,13 @@ int item_pocket::fill_with( const item &contained, int amount, bool allow_unseal
     if( contained_item.charges == 0 ) {
         return 0;
     }
+
+    contained_item.handle_pickup_ownership( const_cast<Character &>( guy ) );
```

Comment:
Why not just pass `Character &guy` into the function instead of const casting here?

Reply:
Thanks for the review.

I may have added const without much thought.
```diff
diff --git a/src/item_pocket.cpp b/src/item_pocket.cpp
index 6c2590b25e..67f878ca6d 100644
--- a/src/item_pocket.cpp
+++ b/src/item_pocket.cpp
@@ -1884,7 +1884,7 @@ bool item_pocket::can_unload_liquid() const
     return will_spill() || !cts_is_frozen_liquid;
 }
 
-int item_pocket::fill_with( const item &contained, const Character &guy, int amount,
+int item_pocket::fill_with( const item &contained, Character &guy, int amount,
                             bool allow_unseal, bool ignore_settings )
 {
     int num_contained = 0;
@@ -1915,7 +1915,7 @@ int item_pocket::fill_with( const item &contained, const Character &guy, int amo
         return 0;
     }
 
-    contained_item.handle_pickup_ownership( const_cast<Character &>( guy ) );
+    contained_item.handle_pickup_ownership( guy );
     if( !insert_item( contained_item ).success() ) {
         debugmsg( "charges per remaining pocket volume does not fit in that very volume" );
         return 0;
diff --git a/src/item_pocket.h b/src/item_pocket.h
index 22e14831f6..a2ece5dc00 100644
--- a/src/item_pocket.h
+++ b/src/item_pocket.h
@@ -325,7 +325,7 @@ class item_pocket
         void add( const item &it, item **ret = nullptr );
         bool can_unload_liquid() const;
 
-        int fill_with( const item &contained, const Character &guy, int amount = 0,
+        int fill_with( const item &contained, Character &guy, int amount = 0,
                        bool allow_unseal = false, bool ignore_settings = false );
 
         /**
```

Is this code better than the old one?